docs/rfcs/sync-manager.md

```diff
@@ -0,0 +1,171 @@
+    * The list of "declined" engines - that is, engines which the user has explicitly declined from syncing. Note that declined engines are global for the account - declining an engine on any device should decline it on every device. While this is something we'd like to consider changing in the future, for now we are sticking with this global declined state.
+    * Some metadata about every collection (which may be subtly different from an "engine"). Currently this metadata is only {version, syncID}
+
+    Management of meta/global includes all syncIDs - if there's no global syncID, or an engine has no syncID, it should generate one. As we mention below, engines will be responsible for persisting the syncID and version and taking necessary action when it changes. Thus, engines never need to communicate a new syncID back to the global.
+
+* Query info/collections and pass this information to engines.
+
+* Manage interaction with the token server to obtain a token. As part of this, the manager needs to detect when an engine fails to use the token, generally due to a node reassignment mid sync. In this scenario, the manager will abort any pending syncs and reset itself, starting a new sync flow. Note also that this means the containing application will generally not need to see authentication errors from engines, only from the manager.
+
+* Manage the "clients" collection - we probably can't ignore this any longer, especially for bookmarks (as desktop will send a wipe command on bookmark restore). This involves both communicating with the engine regarding commands targetting it, and accepting commands to be send to other devices. Note however that it's not yet clear where these commands will originate from - eg, consider a bookmark restore - which point in this process will queue the wipe commands for other clients?). Exactly how to manage this (eg, when we can assume the engine has processed the command?) are TBD.
+
+* Perform, or coordinate, the actual sync of the engines - from the containing app's POV, there's a single "sync now" entry-point. Exactly how the manager binds to engines is TBD. For example, a possible implementation is that a callback is registered with the sync manager. It's also likely that we will expose a way to sync a specific engine rather than "sync all".
+
+* Manage the fact that there may not be a 1:1 relationship between collections and sync engines. For example, it may be true in the future that a single engine uses multiple collections. A possible example is when we come to syncing "containers" - the history engine may then use 2 collections as part of a sync.
+
+## What the sync manager will not do
+
+For completeness, this section lists things which the sync manager will not do:
+
+* It is not a sync scheduler. The containing app will be responsible for knowing when to sync (but as above, the scheduler does know *what* to sync.
+
+(anything else?)
```

Comment:
Hopefully there's quite a bit more it won't do!

But I'm not sure anything else is worth calling out.

Maybe something about telemetry? Or will it do that?

Reply:
* It could manage telemetry aggregation, but not submission. (Up to the app to figure out how to handle the telemetry it collects).
* It doesn't handle FxA authentication, or manage signed in user state. (But it does talk to the token server!)
* It doesn't handle collection validation. (Up to the engines).
* It doesn't track changes. (Up to engines and stores).
* It doesn't make HTTP calls directly, for auth, downloading or uploading. (Up to the token and storage server clients).

Reply:
Thanks - I added more about this and spoke specifically about telemetry.